deploy/crds/pagerduty.openshift.io_pagerdutyintegrations_crd.yaml

```diff
@@ -0,0 +1,136 @@
+                          be empty. This array is replaced during a strategic merge
+                          patch.
+                        items:
+                          type: string
+                        type: array
+                    required:
+                    - key
+                    - operator
+                    type: object
+                  type: array
+                matchLabels:
+                  additionalProperties:
+                    type: string
+                  description: matchLabels is a map of {key,value} pairs. A single
+                    {key,value} in the matchLabels map is equivalent to an element
+                    of matchExpressions, whose key field is "key", the operator is
+                    "In", and the values array contains only "value". The requirements
+                    are ANDed.
+                  type: object
+              type: object
+            escalationPolicy:
+              description: ID of an existing Escalation Policy in PagerDuty.
+              type: string
+            pagerdutyApiKeySecretRef:
+              description: Reference to the secret containing PAGERDUTY_API_KEY.
+              properties:
+                name:
+                  description: Name is unique within a namespace to reference a secret
+                    resource.
+                  type: string
+                namespace:
+                  description: Namespace defines the space within which the secret
+                    name must be unique.
+                  type: string
+              type: object
+            resolveTimeout:
+              description: Time in seconds that an incident is automatically resolved
+                if left open for that long. Value must not be negative. Omitting or
+                setting this field to 0 will disable the feature.
+              minimum: 0
+              type: integer
+            servicePrefix:
+              description: Prefix to set on the PagerDuty Service name.
+              type: string
+            targetSecretRef:
+              description: Name and namespace in the target cluster where the secret
+                is synced.
+              properties:
+                name:
+                  description: Name is unique within a namespace to reference a secret
+                    resource.
+                  type: string
+                namespace:
+                  description: Namespace defines the space within which the secret
+                    name must be unique.
+                  type: string
+              type: object
+          required:
+          - clusterDeploymentSelector
+          - escalationPolicy
+          - pagerdutyApiKeySecretRef
+          - servicePrefix
+          - targetSecretRef
+          type: object
+        status:
+          description: PagerDutyIntegrationStatus defines the observed state of PagerDutyIntegration
+          type: object
```

Comment:
seems your CRD is fine here :)

Reply:
It should be, it's generated from the Go types by the operator-sdk :slightly_smiling_face:

It might be important to hightlight that I _did_make a change after generation though, to work around an issue that Kubernetes/OpenShift v3.11 has with the structured schema stuff that these later versions (0.12+ I believe) add to the CRDs.

The change is made with:

```
yq d -i deploy/crds/pagerduty.openshift.io_pagerdutyintegrations_crd.yaml spec.validation.openAPIV3Schema.type
```

Reply:
this is exactly what I wanted to see. maybe add a `//go:generate` that will regenerate it and do this command?